presto-jdbc/src/main/java/com/facebook/presto/jdbc/ObjectCasts.java

```diff
@@ -0,0 +1,253 @@
+            if (x instanceof Number) {
+                return ((Number) x).longValue();
+            }
+            if (x instanceof String) {
+                return parseLong((String) x);
+            }
+        }
+        catch (RuntimeException e) {
+            throw invalidConversion(x, targetSqlType, e);
+        }
+        throw invalidConversion(x, targetSqlType);
+    }
+
+    public static float castToFloat(Object x, int targetSqlType)
+            throws SQLException
+    {
+        if (x instanceof Boolean) {
+            return (((Boolean) x) ? 1 : 0);
+        }
+        try {
+            if (x instanceof Number) {
+                return ((Number) x).floatValue();
+            }
+            if (x instanceof String) {
+                return parseFloat((String) x);
+            }
+        }
+        catch (RuntimeException e) {
+            throw invalidConversion(x, targetSqlType, e);
+        }
+        throw invalidConversion(x, targetSqlType);
+    }
+
+    public static double castToDouble(Object x, int targetSqlType)
+            throws SQLException
+    {
+        if (x instanceof Boolean) {
+            return (((Boolean) x) ? 1 : 0);
+        }
+        try {
+            if (x instanceof Number) {
+                return ((Number) x).doubleValue();
+            }
+            if (x instanceof String) {
+                return parseDouble((String) x);
+            }
+        }
+        catch (RuntimeException e) {
+            throw invalidConversion(x, targetSqlType, e);
+        }
+        throw invalidConversion(x, targetSqlType);
+    }
+
+    public static BigDecimal castToBigDecimal(Object x, int targetSqlType)
+            throws SQLException
+    {
+        if (x instanceof Boolean) {
+            return BigDecimal.valueOf(((Boolean) x) ? 1 : 0);
+        }
+        if (x instanceof BigInteger) {
+            return new BigDecimal((BigInteger) x);
+        }
+        try {
+            if ((x instanceof Byte) || (x instanceof Short) || (x instanceof Integer) || (x instanceof Long)) {
+                return BigDecimal.valueOf(((Number) x).longValue());
+            }
+            if ((x instanceof Float) || (x instanceof Double)) {
+                return BigDecimal.valueOf(((Number) x).doubleValue());
+            }
+            if (x instanceof String) {
+                return new BigDecimal((String) x);
+            }
+        }
+        catch (RuntimeException e) {
+            throw invalidConversion(x, targetSqlType, e);
+        }
+        throw invalidConversion(x, targetSqlType);
+    }
+
+    public static byte[] castToBinary(Object x, int targetSqlType)
+            throws SQLException
+    {
+        if (x instanceof byte[]) {
+            return (byte[]) x;
+        }
+        try {
+            if (x instanceof String) {
+                return x.toString().getBytes(UTF_8);
+            }
+        }
+        catch (RuntimeException e) {
+            throw invalidConversion(x, targetSqlType, e);
```

Comment:
No, I added it when I read the `toString()` as a generic `Object.toString()` call. I'll remove it now since it can't throw and to save future readers the same question. I only added try/catch in places that could throw. Others, such as `Date` conversion, don't have it.